### PR TITLE
De-duplicate unions.

### PR DESCRIPTION
**Pull request by mitchell-merry**
## Changes

If a union has multiple members that are the same, then we should de-duplicate them.

https://github.com/drwpow/openapi-typescript/pull/1049#discussion_r1159448414

## Checklist

- [x] Unit tests updated
- [ ] README updated
- [x] `examples/` directory updated (if applicable)

### CODE REVIEW

File: examples/github-api-next.ts
Comment (mitchell-merry) [inline examples/github-api-next.ts:91492]: this is pretty nice

**Comment by mitchell-merry** [comment]
Hm, I think this happens to fix https://github.com/drwpow/openapi-typescript/issues/810 as well.

Currently, a schema like this:
```yaml
    MetaQuote:
      oneOf: []
      discriminator:
        propertyName: meta_serialization_key
        mapping: {}
```

generates like so:
```ts
export interface components {
  schemas: {
    MetaQuote: ; // error here
  };
}
```

Now, we default to `never` when no types are passed into `tsUnionOf`, so we get:
```ts
export interface components {
  schemas: {
    MetaQuote: never;
  };
}
```

which I believe is what we expect, right?

File: src/utils.ts
Comment (drwpow) [inline src/utils.ts:213]: ah this is smart! I hadn’t considered this ❤️

File: src/utils.ts
Comment (drwpow) [inline src/utils.ts:219]: 👍

**Review by drwpow — APPROVED** [review]

**Comment by drwpow** [comment]
Really appreciate all your awesome work on this library, and your helpfulness in the issues! I accept your offer to help and have invited you to have maintaner access 🙂. PRs still require approval from another maintainer, but you can at least open PRs and triage issues easier now. Thank you so much! 🙏

**Comment by drwpow** [comment]
When you get accept access to this project, I’ll let you merge this one 🙂

**Pull request merged**
